src/pages/themes/author/[username]/[...page].astro

```diff
@@ -0,0 +1,78 @@
+---
+import CardGrid from '~/components/CardGrid.astro';
+import CardGridGroup from '~/components/CardGridGroup.astro';
+import Pagination from '~/components/Pagination.astro';
+import Profile from '~/components/Profile.astro';
+import SidebarLayout from '~/components/SidebarLayout.astro';
+import SidebarPanel from '~/components/SidebarPanel.astro';
+import { THEMES_API_URL } from '~/helpers/constants.ts';
+import { paginate } from '~/helpers/paginate.js';
+import MainLayout from '~/layouts/MainLayout.astro';
+import SubmitTheme from '../../_components/SubmitTheme.astro';
+import ThemeCard from '../../_components/ThemeCard.astro';
+import type { ThemeAndAuthor } from '../../_types/index.ts';
+
+export const prerender = false;
+
+// with '[...page]' rest routes we'll get undefined for the first page, default that to 1
+// otherwise, try to parse the page number from the URL
+const currentPage =
+	typeof Astro.params.page === 'undefined' ? 1 : Number.parseInt(Astro.params.page);
+
+// invalid page number!
+if (!currentPage || Number.isNaN(currentPage)) {
+	return Astro.redirect('/404');
+}
+
+const allThemes: ThemeAndAuthor[] = await fetch(
+	`${THEMES_API_URL}/api/themes/author?username=${Astro.params.username}`,
```

Comment:
Just noting how we discussed maybe not to use `username` as the param here so we don’t expose GitHub accounts that maybe were not intended to be public like this.

One proposal was to use the numeric author ID instead. Would require updating the API endpoint too.

Reply:
Implemented in this PR. Making another PR to dev portal to implement the API change